endpoints/openrtb2/amp_auction.go

```diff
@@ -453,6 +453,44 @@ func (deps *endpointDeps) overrideWithParams(ampParams amp.Params, req *openrtb2
 		req.TMax = int64(*ampParams.Timeout) - deps.cfg.AMPTimeoutAdjustment
 	}
 
+	if len(ampParams.Targeting) > 0 {
+		err := setTargeting(req, ampParams.Targeting)
+		if err != nil {
+			return []error{err}
+		}
+	}
+
+	return nil
+}
+
+func setTargeting(req *openrtb2.BidRequest, targeting string) error {
+	var targetingData map[string]interface{}
+	if err := json.Unmarshal([]byte(targeting), &targetingData); err != nil {
+		return fmt.Errorf("unable to unmarshal targeting data: %s", err.Error())
+	}
+	_, _, offset, err := jsonparser.Get(req.Imp[0].Ext, "data")
+	if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return errors.New("error parsing imp ext")
+	}
+	if offset != -1 {
+		return errors.New("field 'data' already present in imp ext")
+	}
```

Comment:
I don't think it's worth it to use jsonparser to check for existence of the `data` field when the very next thing we do is unmarshal it. Consider moving the "already present" check after unmarshaling the map.